examples/cluster/plot_cluster_comparison.py

```diff
@@ -35,15 +35,15 @@
 from sklearn.neighbors import kneighbors_graph
 from sklearn.preprocessing import StandardScaler
 
-np.random.seed(0)
-
 # ============
 # Generate datasets. We choose the size big enough to see the scalability
 # of the algorithms, but not too big to avoid too long running times
 # ============
 n_samples = 500
-noisy_circles = datasets.make_circles(n_samples=n_samples, factor=0.5, noise=0.05)
-noisy_moons = datasets.make_moons(n_samples=n_samples, noise=0.05)
+noisy_circles = datasets.make_circles(
+    n_samples=n_samples, factor=0.5, noise=0.05, random_state=8
+)
+noisy_moons = datasets.make_moons(n_samples=n_samples, noise=0.05, random_state=8)
 blobs = datasets.make_blobs(n_samples=n_samples, random_state=8)
 no_structure = np.random.rand(n_samples, 2), None
```

Comment:
This one is not seeded.
So we need to create a random number generator at least for all the `np.random` of the current file:


```suggestion
rng = np.random.RandomState(42)
no_structure = rng.rand(n_samples, 2), None
```

Reply:
Thanks @glemaitre, I have pushed a commit 7dd631f07563c15e16cd30df7168ea86ed430571 that adds the changes you propose.